geodiff/src/drivers/postgresdriver.cpp

```diff
@@ -471,15 +471,16 @@ static bool isColumnInt( const TableColumnInfo &col )
 
 static bool isColumnDouble( const TableColumnInfo &col )
 {
-  return col.type == "real" || col.type == "double precision";
+  return col.type == "real" || col.type == "double precision" || startsWith( col.type.dbType, "numeric" );
 }
 
 static bool isColumnText( const TableColumnInfo &col )
 {
   return col.type == "text" || startsWith( col.type.dbType, "text(" ) ||
          col.type == "varchar" || startsWith( col.type.dbType, "varchar(" ) ||
          col.type == "character varying" || startsWith( col.type.dbType, "character varying(" ) ||
-         col.type == "char" || col.type == "citext";
+         col.type == "char" || startsWith( col.type.dbType, "char(" ) || startsWith( col.type.dbType, "character(" ) ||
+         col.type == "citext" || col.type == "uuid";
```

Comment:
As uuid is not proper text field (and only we treat it as such for now), I think my preference would be not to have it listed in `isColumnText()` function, and rather in `resultToValue()` we would do something like:
```cpp
    else if ( isColumnText( col ) || col.type == "uuid" ) { ... }
```